internal/storage/dropbox/dropbox.go

```diff
@@ -0,0 +1,182 @@
+package dropbox
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox"
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox/files"
+	"github.com/offen/docker-volume-backup/internal/storage"
+)
+
+type dropboxStorage struct {
+	*storage.StorageBackend
+	client files.Client
+}
+
+// Config allows to configure a Dropbox storage backend.
+type Config struct {
+	Token      string
+	RemotePath string
+}
+
+// NewStorageBackend creates and initializes a new Dropbox storage backend.
+func NewStorageBackend(opts Config, logFunc storage.Log) (storage.Backend, error) {
+	if opts.Token == "" {
```

Comment:
Considering the top level func will not even try to call this when there's no token I could live without this check personally.